Software/app/freefall_lorawan/lora_app.c

```diff
@@ -0,0 +1,169 @@
+/**
+  ******************************************************************************
+  * @file    lora_app.c
+  * @author  MCD Application Team
+  * @brief   Application of the LRWAN Middleware
+  ******************************************************************************
+  * @attention
+  *
+  * <h2><center>&copy; Copyright (c) 2020 STMicroelectronics.
+  * All rights reserved.</center></h2>
+  *
+  * This software component is licensed by ST under Ultimate Liberty license
+  * SLA0044, the "License"; You may not use this file except in compliance with
+  * the License. You may obtain a copy of the License at:
+  *                             www.st.com/SLA0044
+  *
+  ******************************************************************************
+  */
+
+#include "app.h"
+#include "Region.h" /* Needed for LORAWAN_DEFAULT_DATA_RATE */
+#include "stm32_timer.h"
+#include "sys_app.h"
+#include "lora_app.h"
+#include "stm32_seq.h"
+#include "stm32_lpm.h"
+#include "LmHandler.h"
+
+/**
+  * @brief  join event callback function
+  * @param  params
+  * @return none
+  */
+static void OnJoinRequest(LmHandlerJoinParams_t *joinParams);
+
+/**
+  * @brief  tx event callback function
+  * @param  params
+  * @return none
+  */
+static void OnTxData(LmHandlerTxParams_t *params);
+
+/*!
+ * Will be called each time a Radio IRQ is handled by the MAC layer
+ *
+ */
+static void OnMacProcessNotify(void);
+
+/*!
+ * User application buffer
+ */
+static uint8_t AppDataBuffer[LORAWAN_APP_DATA_BUFFER_MAX_SIZE];
+
+/*!
+ * User application data structure
+ */
+static LmHandlerAppData_t AppData = {0, 0, AppDataBuffer};
+
+static ActivationType_t ActivationType = LORAWAN_DEFAULT_ACTIVATION_TYPE;
+
+/*!
+ * LoRaWAN handler Callbacks
+ */
+static LmHandlerCallbacks_t LmHandlerCallbacks =
+    {
+        .GetBatteryLevel = GetBatteryLevel,
+        .GetTemperature = GetTemperatureLevel,
+        .GetUniqueId = GetUniqueId,
+        .GetRandomSeed = GetRandomSeed,
+        .OnJoinRequest = OnJoinRequest,
+        .OnTxData = OnTxData,
+        .OnMacProcess = OnMacProcessNotify};

```

Comment:
Please bring back `.OnRxData = OnRxData`, and the function should just do this
```c
static void OnRxData(LmHandlerAppData_t *appData, LmHandlerRxParams_t *params)
{
/**
* User can add handling of Downlink data here
*/
 static const char *slotStrings[] = {"1", "2", "C", "C Multicast", "B Ping-Slot", "B Multicast Ping-Slot"};
  APP_LOG(TS_OFF, VLEVEL_M, "\r\n ###### D/L FRAME:%04d | SLOT:%s | PORT:%d | DR:%d | RSSI:%d | SNR:%d\r\n",
            params->DownlinkCounter, slotStrings[params->RxSlot], appData->Port, params->Datarate, params->Rssi, params->Snr);
    switch (appData->Port)
APP_LOG(TS_OFF, VLEVEL_M, "\r\n Unhandled Downlink, Received %d bytes on port: %d \r\n", appData->BufferSize, appData->Port);
}
```

Reply:
This produces this error message:
> ../../app/freefall_lorawan/lora_app.c: In function 'OnRxData':
../../app/freefall_lorawan/lora_app.c:159:105: warning: statement will never be executed [-Wswitch-unreachable]
  159 |         APP_LOG(TS_OFF, VLEVEL_M, "\r\n Unhandled Downlink, Received %d bytes on port: %d \r\n", appData->BufferSize, appData->Port);

Reply:
Just remove `  switch (appData->Port)`, I added this on the fly